rnn/training.ipynb

```diff
@@ -0,0 +1,1408 @@
+      },
+      "source": [
+        "# Preparing the Data"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ePqZvenLH7mn",
+        "colab_type": "text"
+      },
+      "source": [
+        "We mount our Google Drive to access the data files from within this notebook."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "QrX1HvAWH2xM",
+        "colab_type": "code",
+        "outputId": "80f361d7-7bc4-42d5-f1a5-0d91303a6338",
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 189
+        }
+      },
+      "source": [
+        "from google.colab import drive\n",
+        "\n",
+        "mounted_folder_fp = '/content/drive'\n",
+        "drive.mount(mounted_folder_fp, force_remount=True)"
+      ],
+      "execution_count": 3,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code\n",
+            "\n",
+            "Enter your authorization code:\n",
+            "··········\n",
+            "Mounted at /content/drive\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "UhzHmMNjOcB6",
+        "colab_type": "text"
+      },
+      "source": [
+        "We then load all the JSON files we need from the mounted drive."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "MMuVqd2espNQ",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "with open('{}/My Drive/Colab Notebooks/hola/data/cmudict-char-ids.json'.format(mounted_folder_fp), 'r', encoding='ascii') as f:\n",
+        "    char_ids_dict = json.load(f)\n",
+        "\n",
+        "with open('{}/My Drive/Colab Notebooks/hola/data/cmudict-phoneme-ids.json'.format(mounted_folder_fp), 'r', encoding='ascii') as f:\n",
+        "    phoneme_ids_dict = json.load(f)\n",
+        "\n",
+        "with open('{}/My Drive/Colab Notebooks/hola/data/cmudict-processed.json'.format(mounted_folder_fp), 'r', encoding='ascii') as f:\n",
+        "    cmudict_pairs = json.load(f)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "73jb0ZTnfgIF",
+        "colab_type": "text"
+      },
+      "source": [
+        "We find the length of the longest word/list of phonemes to define the length of our output PyTorch tensors later."
```

Comment:
yeah, it will cause an error

we need to manually define a vector length cause all words need to be padded to that length, which usually is the length of the longest word (tho we could manually define it to be something bigger if we wanted)